{{cookiecutter.repo_name}}/.github/workflows/CI.yaml

```diff
@@ -0,0 +1,74 @@
+name: CI
+
+on: [push]
+
+jobs:
+  test:
+
+    runs-on: {{ '${{ matrix.os }}' }}
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+        python-version: [3.6, 3.7]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Additional info about the build
+      shell: bash
+      run: |
+        uname -a
+        df -h
+        ulimit -a
+
+{% if cookiecutter.dependency_source != 'Dependencies from pip only (no conda)' %}
+    - name: Setup conda (Ubuntu, macOS)
+      shell: bash
+      if: startsWith(matrix.os, 'macOS') || startsWith(matrix.os, 'ubuntu')
+      run: echo "::set-env name=CONDA_EXE::$CONDA/bin/conda"
+    - name: Setup conda (Windows)
+      shell: bash
+      if: startsWith(matrix.os, 'windows')
+      run: echo "::set-env name=CONDA_EXE::$CONDA/Scripts/conda.exe"
+{% endif %}
```

Comment:
Please do look through this YAML, a lot of this simply isn't needed: https://github.com/MolSSI/QCElemental/blob/827e3e6faf9ba08f327bafff87e893b8f8d559fa/.github/workflows/CI.yml

Reply:
So `conda` is in `PATH`? Cool!

Reply:
Looks like this is not applicable to Windows! Shall I revert the changes?

Reply:
Maybe add it to the path for Windows only?

Reply:
I followed [this discussion](https://github.com/actions/starter-workflows/issues/44#issuecomment-554649325) and there seems to be some problems with path modification. I'll try to use `condabin` instead, and see if that works better.

Reply:
Actually, [looks like](https://github.com/openkinome/kinoml/pull/7/checks?check_run_id=328385167) `conda` is only in `PATH` in Ubuntu. I keep thinking that the `CONDA_EXE` approach is the cleanest although it requires one more setup step.

We could use a single step too:

```
- name: Set-up conda
        run:   |
          if [ "${{ matrix.os }}" != "windows-latest" ]; then
            echo '::set-env name=CONDA_EXE::${CONDA}/bin/conda'
          else
            echo '::set-env name=CONDA_EXE::C:\Miniconda\condabin\conda.bat'
          fi
        shell: bash
```

Reply:
Do you think we could add a conda function in this manner? If not I would recommend something simple so that further lines can do `eval '$CONDA_INIT'` or similar.

Reply:
Best thing would be modifying the `bash_profile`/`bashrc` to autostart conda for us (but we should still run `conda activate test` once the environment has been created, although we could check for `test` existence and activate that instead?), but I think that default `bash` interpreters in GitHub are [configured to ignore profile files](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#using-a-specific-shell), so we would need to define a custom `shell` in each step. I am not aware of any option to override the default `shell` for all steps - if that's possible, it'd be the best way. Anyway, shall I try with the `bashrc`+`custom bash shell` approach?

Reply:
Ok, I've tried this route but it's not pretty at all because we would need to _always_ specify `shell: bash --noprofile --rcfile conda_init.sh -i -e -o pipefail {0}`. I find that very confusing and hard to reason about. 

Instead, I have opted to create [this auxiliary script](https://github.com/openkinome/kinoml/blob/329f764b780e9bcfa228a972625ab3ee7e840729/devtools/scripts/initialize_conda.sh) and call it at the beginning of each step:

```
    - name: Install package
      shell: bash
      run: |
        . devtools/scripts/initialize_conda.sh
        python -m pip install . --no-deps
        conda list
```

That way the intention is clear and the initialization code is easier to maintain. Would you prefer this approach?

Reply:
I went ahead and added the script in the last commit.